### PR TITLE
Support batchCreate.

### PR DESCRIPTION
**Pull request by yuchenshi**
### Description

This adds support for batchCreate, with a few corner cases intentionally omitted for sanity. One step closer to import/export!

### Scenarios Tested

Comprehensive spec test added.

### CODE REVIEW

File: src/emulator/auth/operations.ts
Comment (yuchenshi) [inline src/emulator/auth/operations.ts:319]: This is not the production error message. I'll add a test and fix this before I merge.

File: src/emulator/auth/operations.ts
Comment (yuchenshi) [inline src/emulator/auth/operations.ts:345]: TODO: Production catches this case only if `sanityCheck && emailRequired`. We should use production error message in that case, otherwise throw this emulator specific error message (since we do not want to support this edge case)

**Review by samtstern — APPROVED** [review]
LGTM assuming you resolve your own TODOs

**Pull request merged**
